src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -844,6 +848,35 @@ DeferredErrorIfUnsupportedRecurringTuplesJoin(
 				break;
 			}
 		}
+		else if (joinType == JOIN_INNER && plannerInfo->hasLateralRTEs)
+		{
+			bool innerrelOnlyRecurring = RelationInfoContainsOnlyRecurringTuples(
+				plannerInfo, innerrelRelids);
+			bool outerrelOnlyRecurring = RelationInfoContainsOnlyRecurringTuples(
+				plannerInfo, outerrelRelids);
+			if (innerrelOnlyRecurring && !outerrelOnlyRecurring)
```

Comment:
this might error on <dist>, LATERAL (SELECT .. <ref>) type queries, which work fine so we probably don't want to break those

Reply:
There's this query that looks fine, but also fails
```sql
SELECT count(*)
FROM test,
    LATERAL (
        SELECT
            test_2.y
        FROM test test_2
        WHERE
            test_2.x = test.x
        LIMIT 2
    ) q
    JOIN ref ON q.y = ref.a;
```